src/CommunityToolkit.Maui/Views/RatingView/RatingView.shared.cs

```diff
@@ -0,0 +1,748 @@
+﻿// Ignore Spelling: color, bindable, colors
+
+using System.Collections.ObjectModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Windows.Input;
+using CommunityToolkit.Maui.Core;
+using Microsoft.Maui.Controls.Shapes;
+
+namespace CommunityToolkit.Maui.Views;
+
+/// <summary>Rating view control.</summary>
+public class RatingView : TemplatedView, IRatingView
+{
+	/// <summary>Bindable property for <see cref="CustomItemShape"/> bindable property.</summary>
+	public static readonly BindableProperty CustomItemShapeProperty = RatingViewItemElement.CustomShapeProperty;
+
+	/// <summary>The backing store for the <see cref="EmptyColor" /> bindable property.</summary>
+	public static readonly BindableProperty EmptyColorProperty = RatingViewItemElement.EmptyColorProperty;
+
+	/// <summary>The backing store for the <see cref="FilledColor" /> bindable property.</summary>
+	public static readonly BindableProperty FilledColorProperty = RatingViewItemElement.FilledColorProperty;
+
+	/// <summary>The backing store for the <see cref="IsReadOnly" /> bindable property.</summary>
+	public static readonly BindableProperty IsReadOnlyProperty = BindableProperty.Create(nameof(IsReadOnly), typeof(bool), typeof(RatingView), defaultValue: RatingViewDefaults.IsReadOnly, propertyChanged: OnIsReadOnlyChanged);
+
+	/// <summary>Bindable property for <see cref="ItemPadding"/> bindable property.</summary>
+	public static readonly BindableProperty ItemPaddingProperty = RatingViewItemElement.ItemPaddingProperty;
+
+	// TODO: Add some kind of Roslyn Analyser to validate at design/build time that the MaximumRating is in bounds (1-25).
```

Comment:
Why is our maximum bound set to 25? Is this a platform-specific problem? 

Or are we just picking an arbitrary number for a maximum? 

I'm totally fine with us setting a maximum value, but I'm just curious if it's a platform constraint or an artificial constraint set by us?